bin/s2nc.c

```diff
@@ -535,16 +571,9 @@ int main(int argc, char *const *argv)
 
         /* Save session state from connection if reconnect is enabled */
         if (reconnect > 0) {
-            if (!session_ticket && s2n_connection_get_session_id_length(conn) <= 0) {
-                printf("Endpoint sent empty session id so cannot resume session\n");
-                exit(1);
-            }
-            free(session_state);
-            session_state_length = s2n_connection_get_session_length(conn);
-            session_state = calloc(session_state_length, sizeof(uint8_t));
-            if (s2n_connection_get_session(conn, session_state, session_state_length) != session_state_length) {
-                print_s2n_error("Error getting serialized session state");
-                exit(1);
+            /* Only need to wait to receive the session ticket in TLS1.3 */
+            if ((conn->actual_protocol_version >= S2N_TLS13 ) && session_ticket) {
+                GUARD_EXIT(echo(conn, sockfd, &session_ticket_recv), "Error calling echo");
```

Comment:
This is a continuation on my previous comment: 
The two outcomes for echo are `success: return 0` and  `failure: abort with exit(1)` While using 
GUARD is the preferred choice and not wrong here, it is not really useful  with `echo` because of this behavior as we will never get a return value less than 0.

```
#define GUARD_EXIT(x, msg)  \
  do {                      \
    if ((x) < 0) {          \
      print_s2n_error(msg); \
      exit(1);              \
    }                       \
  } while (0)
```